pkg/webhook/validator.go

```diff
@@ -1012,3 +1038,105 @@ func refOrFieldError(image, field string, index int) (name.Reference, *apis.Fiel
 	}
 	return ref, nil
 }
+
+// configFileResult is used to communicate results from gofuncs that fetch
+// ConfigFiles for a given image.
+// Because this can be recursive (say, multi-arch image), returns a map where
+// key is the architecture of the container image.
+type configFileResult struct {
+	ret  map[string]*v1.ConfigFile
+	errs []error
+}
+
+// getConfigs will fetch ConfigFile(s) for a given image. In case the image
+// is an index, we'll fetch the arch images recursively.
+func getConfigs(ctx context.Context, ref name.Reference, options ...remote.Option) (map[string]*v1.ConfigFile, []error) {
+	descriptor, err := remote.Get(ref, options...)
+	if err != nil {
+		return nil, []error{fmt.Errorf("failed to get ref %s : %w", ref.String(), err)}
+	}
+	switch descriptor.MediaType {
+	case types.OCIImageIndex, types.DockerManifestList:
+		ii, err := descriptor.ImageIndex()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting ImageIndex for %s : %w", ref.String(), err)}
+		}
+		im, err := ii.IndexManifest()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting IndexManifest for %s : %w", ref.String(), err)}
+		}
+		wg := new(sync.WaitGroup)
+
+		results := make(chan configFileResult, len(im.Manifests))
+		for _, manifest := range im.Manifests {
+			manifest := manifest
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				newRefString := ref.Context().Digest(manifest.Digest.String()).String()
+				newRef, err := name.ParseReference(newRefString)
+				if err != nil {
+					results <- configFileResult{ret: nil, errs: []error{fmt.Errorf("failed to ParseReference for: %s: %w", newRefString, err)}}
+					return
+				}
+
+				newRefConfigs, errs := getConfigs(ctx, newRef)
+				results <- configFileResult{ret: newRefConfigs, errs: errs}
+			}()
+		}
+		errs := []error{}
+		ret := make(map[string]*v1.ConfigFile, len(im.Manifests))
+		for i := 0; i < len(im.Manifests); i++ {
+			select {
+			case <-ctx.Done():
+				errs = append(errs, errors.New("context canceled"))
+			case result, ok := <-results:
+				if !ok {
+					errs = append(errs, errors.New("channel closed before all results were gathered"))
+				} else {
+					if result.errs != nil {
+						errs = append(errs, fmt.Errorf("failed to get a ConfigFile: %w", err))
+					} else {
+						for k, v := range result.ret {
+							ret[k] = v
+						}
+					}
+				}
+			}
+		}
+		wg.Wait()
+		if len(errs) > 0 {
+			return nil, errs
+		}
+		return ret, nil
+	case types.OCIManifestSchema1, types.DockerManifestSchema2:
+		// This is an Image, so just return it.
+		image, err := descriptor.Image()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting Image for %s: %w", ref.String(), err)}
+		}
+		cf, err := image.ConfigFile()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting ConfigFile for %s: %w", ref.String(), err)}
+		}
+		return map[string]*v1.ConfigFile{normalizeArchitecture(cf): cf}, nil
+	default:
```

Comment:
Q: I am wondering if we should error if there are schema 1 images `types.DockerManifestSchema1, types.DockerManifestSchema1Signed` ? Or should we just warn ?

Reply:
My understanding is that they've been deprecated for looong time now and support will be removed real soon now. 
This is also how cosign deals with them:
https://github.com/sigstore/cosign/blob/main/pkg/oci/mutate/map.go#L126